modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/throttling/SpeedBasedMemoryConsumptionThrottlingStrategy.java

```diff
@@ -496,5 +506,35 @@ void reset() {
     void finish() {
         cpWriteSpeed.closeInterval();
         threadIds.clear();
+
+        updateWriteVsFsyncCoefficient();
+    }
+
+    private void updateWriteVsFsyncCoefficient() {
+        CheckpointProgress progress = cpProgress.get();
+        assert progress != null;
+
+        if (progress.currentCheckpointPagesCount() == 0) {
+            return;
+        }
+
+        long pagesWriteTimeMillis = progress.getPagesWriteTimeMillis();
+        long fsyncTimeMillis = progress.getFsyncTimeMillis();
+
+        double coefficient = ((double) pagesWriteTimeMillis) / (pagesWriteTimeMillis + fsyncTimeMillis);
+        if (isNaN(coefficient)) {
+            return;
+        }
+
+        double newCoefficient = writeVsFsyncCoefficient * 0.85 + coefficient * 0.15;
```

Comment:
What's this about?

Reply:
That's an exponential smoothing of the coefficient